### ISSUE TITLE
Documentation: tool options are inherited

### ISSUE THREAD

**Issue by imphil** [olofk/fusesoc#504]
The tools section is mentioned as not inherited in https://fusesoc.readthedocs.io/en/latest/user/build_system/dependencies.html#what-happens-if-a-dependency-is-specified which is wrong. Also double-check the other sections and maybe add a test?

**Comment by olofk** [comment]
Good catch. I just checked and the rest of them looks correct. Tests would be good though
